src/OrleansServiceBus/Providers/Streams/EventHub/EventHubCheckpointer.cs

```diff
@@ -3,17 +3,14 @@
 using System.Threading.Tasks;
 using Microsoft.ServiceBus.Messaging;
 using Orleans.AzureUtils;
+using Orleans.Streams;
 
 namespace Orleans.ServiceBus.Providers
 {
     /// <summary>
-    /// This class stores EventHub partition checkpoint information (a partition offset) in azure table storage.
-    /// TODO: Use dependency injection to fill this behavior.
-    /// This class stores data into table storage, which will not be ideal for all users.  Further, it introduces
-    ///   a dependency on azure storage to this assembly.  Ideally this should be an injectable behavior, rather
-    ///   than a concrete class.
+    /// This class stores EventHub partition checkpointer information (a partition offset) in azure table storage.
     /// </summary>
-    internal class EventHubPartitionCheckpoint
+    public class EventHubCheckpointer : IStreamQueueCheckpointer<string>
```

Comment:
how about `AzureTableBasedEventHubCheckpointer`?
On reading further, what makes this EventHub specific? this could just be `AzureTableBasedStreamQueueCheckpointer` or something like that. `EventHubPartitionCheckpointEntity` should probably be renamed too, since it's not EH specific either... Although the only specific thing I see is the usage of the `EventHubConsumerGroup.StartOfStream` constant